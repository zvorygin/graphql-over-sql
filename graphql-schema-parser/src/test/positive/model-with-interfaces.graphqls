schema {
    query: Query
}

typeReference Query {
    personById(id: [Int!]!): [IPerson!]!
    customerById(id: [Int!]!): [ICustomer!]!
    employeeById(id: [Int!]!): [IEmployee!]!
}

#@Provider(name="sql")
#@Table(name="PERSON", discover=true)
interface IPerson {
#    #@Column("PERSON_NUMBER")
#    #@Id
    id: Int!
}

#@Provider(name="sql")
#@Table(name="CUSTOMER", discover=true)
#@Parent(name="IPerson")
interface ICustomer {
    id: Int!
}

#@Provider(name="sql")
#@Table(name="EMPLOYEE", discover=true)
#@Parent(name="IPerson")
interface IEmployee {
    id: Int!
    email: String!
}

interface IUser {
    email: String!
    ldap: String
    hiredAt: String
}

typeReference Customer implements ICustomer, IUser {

}

typeReference Employee implements IEmployee {

}