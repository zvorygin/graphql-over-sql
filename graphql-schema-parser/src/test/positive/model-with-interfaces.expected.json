{
  "schema" : {
    "queryType" : {
      "collection" : false,
      "nonNull" : false,
      "typeName" : "Query"
    }
  },
  "types" : {
    "Employee" : {
      "interfaces" : [ "IEmployee" ],
      "name" : "Employee"
    },
    "Query" : {
      "fields" : [ {
        "arguments" : {
          "id" : {
            "argumentName" : "id",
            "typeReference" : {
              "collection" : false,
              "nonNull" : true,
              "typeName" : "Int",
              "wrappedType" : {
                "collection" : true,
                "nonNull" : false,
                "typeName" : "Int",
                "wrappedType" : {
                  "collection" : false,
                  "nonNull" : true,
                  "typeName" : "Int",
                  "wrappedType" : {
                    "collection" : false,
                    "nonNull" : false,
                    "typeName" : "Int"
                  }
                }
              }
            }
          }
        },
        "name" : "customerById",
        "type" : {
          "collection" : false,
          "nonNull" : true,
          "typeName" : "ICustomer",
          "wrappedType" : {
            "collection" : true,
            "nonNull" : false,
            "typeName" : "ICustomer",
            "wrappedType" : {
              "collection" : false,
              "nonNull" : true,
              "typeName" : "ICustomer",
              "wrappedType" : {
                "collection" : false,
                "nonNull" : false,
                "typeName" : "ICustomer"
              }
            }
          }
        }
      }, {
        "arguments" : {
          "id" : {
            "argumentName" : "id",
            "typeReference" : {
              "collection" : false,
              "nonNull" : true,
              "typeName" : "Int",
              "wrappedType" : {
                "collection" : true,
                "nonNull" : false,
                "typeName" : "Int",
                "wrappedType" : {
                  "collection" : false,
                  "nonNull" : true,
                  "typeName" : "Int",
                  "wrappedType" : {
                    "collection" : false,
                    "nonNull" : false,
                    "typeName" : "Int"
                  }
                }
              }
            }
          }
        },
        "name" : "employeeById",
        "type" : {
          "collection" : false,
          "nonNull" : true,
          "typeName" : "IEmployee",
          "wrappedType" : {
            "collection" : true,
            "nonNull" : false,
            "typeName" : "IEmployee",
            "wrappedType" : {
              "collection" : false,
              "nonNull" : true,
              "typeName" : "IEmployee",
              "wrappedType" : {
                "collection" : false,
                "nonNull" : false,
                "typeName" : "IEmployee"
              }
            }
          }
        }
      }, {
        "arguments" : {
          "id" : {
            "argumentName" : "id",
            "typeReference" : {
              "collection" : false,
              "nonNull" : true,
              "typeName" : "Int",
              "wrappedType" : {
                "collection" : true,
                "nonNull" : false,
                "typeName" : "Int",
                "wrappedType" : {
                  "collection" : false,
                  "nonNull" : true,
                  "typeName" : "Int",
                  "wrappedType" : {
                    "collection" : false,
                    "nonNull" : false,
                    "typeName" : "Int"
                  }
                }
              }
            }
          }
        },
        "name" : "personById",
        "type" : {
          "collection" : false,
          "nonNull" : true,
          "typeName" : "IPerson",
          "wrappedType" : {
            "collection" : true,
            "nonNull" : false,
            "typeName" : "IPerson",
            "wrappedType" : {
              "collection" : false,
              "nonNull" : true,
              "typeName" : "IPerson",
              "wrappedType" : {
                "collection" : false,
                "nonNull" : false,
                "typeName" : "IPerson"
              }
            }
          }
        }
      } ],
      "name" : "Query"
    },
    "Customer" : {
      "interfaces" : [ "ICustomer", "IUser" ],
      "name" : "Customer"
    }
  },
  "interfaces" : {
    "IPerson" : {
      "annotations" : {
        "Table" : {
          "attributes" : {
            "discover" : true,
            "name" : "PERSON"
          },
          "name" : "Table"
        },
        "Provider" : {
          "attributes" : {
            "name" : "sql"
          },
          "name" : "Provider"
        }
      },
      "fields" : [ {
        "name" : "id",
        "type" : {
          "collection" : false,
          "nonNull" : true,
          "typeName" : "Int",
          "wrappedType" : {
            "collection" : false,
            "nonNull" : false,
            "typeName" : "Int"
          }
        }
      } ],
      "name" : "IPerson"
    },
    "IEmployee" : {
      "annotations" : {
        "Table" : {
          "attributes" : {
            "discover" : true,
            "name" : "EMPLOYEE"
          },
          "name" : "Table"
        },
        "Parent" : {
          "attributes" : {
            "name" : "IPerson"
          },
          "name" : "Parent"
        },
        "Provider" : {
          "attributes" : {
            "name" : "sql"
          },
          "name" : "Provider"
        }
      },
      "fields" : [ {
        "name" : "id",
        "type" : {
          "collection" : false,
          "nonNull" : true,
          "typeName" : "Int",
          "wrappedType" : {
            "collection" : false,
            "nonNull" : false,
            "typeName" : "Int"
          }
        }
      }, {
        "name" : "email",
        "type" : {
          "collection" : false,
          "nonNull" : true,
          "typeName" : "String",
          "wrappedType" : {
            "collection" : false,
            "nonNull" : false,
            "typeName" : "String"
          }
        }
      } ],
      "name" : "IEmployee"
    },
    "ICustomer" : {
      "annotations" : {
        "Table" : {
          "attributes" : {
            "discover" : true,
            "name" : "CUSTOMER"
          },
          "name" : "Table"
        },
        "Parent" : {
          "attributes" : {
            "name" : "IPerson"
          },
          "name" : "Parent"
        },
        "Provider" : {
          "attributes" : {
            "name" : "sql"
          },
          "name" : "Provider"
        }
      },
      "fields" : [ {
        "name" : "id",
        "type" : {
          "collection" : false,
          "nonNull" : true,
          "typeName" : "Int",
          "wrappedType" : {
            "collection" : false,
            "nonNull" : false,
            "typeName" : "Int"
          }
        }
      } ],
      "name" : "ICustomer"
    },
    "IUser" : {
      "fields" : [ {
        "name" : "ldap",
        "type" : {
          "collection" : false,
          "nonNull" : false,
          "typeName" : "String"
        }
      }, {
        "name" : "hiredAt",
        "type" : {
          "collection" : false,
          "nonNull" : false,
          "typeName" : "String"
        }
      }, {
        "name" : "email",
        "type" : {
          "collection" : false,
          "nonNull" : true,
          "typeName" : "String",
          "wrappedType" : {
            "collection" : false,
            "nonNull" : false,
            "typeName" : "String"
          }
        }
      } ],
      "name" : "IUser"
    }
  },
  "scalars" : {
    "Float" : {
      "name" : "Float"
    },
    "String" : {
      "name" : "String"
    },
    "Boolean" : {
      "name" : "Boolean"
    },
    "Int" : {
      "name" : "Int"
    }
  },
  "inputs" : { }
}